Sources/SemanticAnalyzer/TypeChecker.swift

```diff
@@ -297,7 +297,28 @@ public struct TypeChecker: ASTPass {
   }
 
   public func postProcess(subscriptExpression: SubscriptExpression, passContext: ASTPassContext) -> ASTPassResult<SubscriptExpression> {
```

Comment:
Any reason why this code isn't in the `process` step rather than `postProcess`? Code should preferably be in `process`.

Reply:
There was a reason with how I was planning to do it at the start before I realised that the array identifier was included in the subscript expression. I'll make the change now.